pylint/checkers/refactoring/implicit_booleaness_checker.py

```diff
@@ -209,9 +209,17 @@ def _check_use_implicit_booleaness_not_comparison(
                     )
 
     @staticmethod
-    def base_classes_of_node(instance: nodes.ClassDef) -> List[str]:
-        """Return all the classes names that a ClassDef inherit from including 'object'."""
-        try:
-            return [instance.name] + [x.name for x in instance.ancestors()]
-        except TypeError:
-            return [instance.name]
+    def base_classes_and_return_names(
```

Comment:
I feel like we should have two functions, one for unbound method (``base_classes_of_returned_instance``) and one for instances (``base_classes_of_instance``). This new function is handling two cases.

Reply:
Not sure.. How would we then call the function that calls `base_classes_of_returned_instance` and `base_classes_of_instance`?

Reply:
Do we need one ? It seems the code before was not handling returned instances ?

Reply:
Yeah, that's what the crash is about. We don't only get `Instances` here but also `UnboundMethod` whenever one of the compared values is a `property`. 

This is basically three changes in one: fix method typing to actually reflect that is handling `Instances` + update method to also handle `UnboundMethod` + update method to also handle `Uninsurable` (and everything else that is not `Instance` or `UnboundMethod`. Perhaps I should have split this, but felt that this made more sense.

Reply:
Ok, I understand now. Maybe the original name, "base class of node" is generic enough (and future proof if we encounter exotic codebase that will have other nodes) ?

Reply:
Hmm, for return values we're not really returning "base classes". We're returning the name of their inferred value. That's why I initially changed it.

Reply:
Right. Maybe ``names_of_base_classes_of_node`` ?

Reply:
Not really a big fan of using `classes` as that links so directly with `ClassDef` or `Instance` and this function can (and should) do more. But perhaps that's just because that was what was wrong before and I'm now biased against it.

I can't really think of anything better myself. I'll try again tomorrow and otherwise I'll use that!

Reply:
`name_of_base_classes` is the return, `node` is the input I don't know how to make this clearer though.

Reply:
Yeah but when the return value of a `UnboundMethod` is a `nodes.Dict` and we return `nodes.Dict.name` it doesn't really make sense to refer to `nodes.Dict.name` as the name of the base class of the return value. Oh well, it's just semantics no reason to worry about this too long.

Reply:
I get what you mean now. Maybe calling one of the two functions using the result of ``isinstance`` makes more sense so it's easier to understand what is happening.

Reply:
Yeah perhaps, but then it feels like we're creating a function just because we can't think of a better name 😄 I'll try and come up with something!

Reply:
Renamed to `base_names_of_base_node`

Reply:
Neat !

Reply:
This comment was outdated when you did your review. Are you also okay with the new new name?

Reply:
Yes :)

Reply:
Then I'll merge 🎉